src/nodedef.cpp

```diff
@@ -618,6 +623,18 @@ void ContentFeatures::deSerialize(std::istream &is)
 		if (is.eof())
 			throw SerializationError("");
 		alpha = static_cast<enum AlphaMode>(tmp);
+
+		tmp = readU8(is);
+		if (is.eof()) {
+			move_resistance = liquid_viscosity;
```

Comment:
The `move_resistance = liquid_viscosity;` should happen further up instead of mixing this into the serialization code here IMO